src/common/wui_api.c

```diff
@@ -0,0 +1,282 @@
+
+    ip4addr_ntoa_r(&(config->lan.addr_ip4), addr, IP4_ADDR_STR_SIZE);
+    ip4addr_ntoa_r(&(config->lan.msk_ip4), msk, IP4_ADDR_STR_SIZE);
+    ip4addr_ntoa_r(&(config->lan.gw_ip4), gw, IP4_ADDR_STR_SIZE);
+
+    snprintf(dest, MAX_INI_SIZE,
+        "[lan_ip4]\ntype=%s\nhostname=%s\naddress=%s\nmask=%s\ngateway=%s\n",
+        IS_LAN_STATIC(config->lan.flag) ? "STATIC" : "DHCP", config->hostname,
+        addr, msk, gw);
+}
+
+void stringify_eth_for_screen(char * dest, ETH_config_t * config) {
+    char addr[IP4_ADDR_STR_SIZE], msk[IP4_ADDR_STR_SIZE], gw[IP4_ADDR_STR_SIZE], mac[MAC_ADDR_STR_LEN];
+
+    ip4addr_ntoa_r(&(config->lan.addr_ip4), addr, IP4_ADDR_STR_SIZE);
+    ip4addr_ntoa_r(&(config->lan.msk_ip4), msk, IP4_ADDR_STR_SIZE);
+    ip4addr_ntoa_r(&(config->lan.gw_ip4), gw, IP4_ADDR_STR_SIZE);
+    parse_MAC_address(mac);
+
+    snprintf(dest, 150, "IPv4 Address:\n  %s      \nIPv4 Netmask:\n  %s      \nIPv4 Gateway:\n  %s      \nMAC Address:\n  %s",
+        addr, msk, gw, mac);
+}
+
+void update_eth_addrs(ETH_config_t *config){
+    config->var_mask = ETHVAR_MSK(ETHVAR_LAN_FLAGS);
+    load_eth_params(config);
+
+    if (IS_LAN_STATIC(config->lan.flag)){
+        config->var_mask = ETHVAR_STATIC_LAN_ADDRS;
+        load_eth_params(config);
+    } else {
+        get_addrs_from_dhcp(config);
+    }
+}
+
+uint32_t set_loaded_eth_params(ETH_config_t * config){
+    
+    
+    if (config->var_mask & ETHVAR_MSK(ETHVAR_LAN_FLAGS)) {
+        // if lan type is set to STATIC
+        if (IS_LAN_STATIC(config->lan.flag)){
+            if ((config->var_mask & ETHVAR_STATIC_LAN_ADDRS) != ETHVAR_STATIC_LAN_ADDRS) {
+                return 0;
+            }
+        }
+    }
+    if (config->var_mask & ETHVAR_MSK(ETHVAR_HOSTNAME)) {
+        strlcpy(eth_hostname, config->hostname, ETH_HOSTNAME_LEN + 1);
+    }
+
+    // ugly way to aquire lan flags before load
+    uint8_t swaper, prev_lan_flag = config->lan.flag;
+    uint32_t set_mask = config->var_mask;
+    config->var_mask = ETHVAR_MSK(ETHVAR_LAN_FLAGS);
+    load_eth_params(config);
+    swaper = prev_lan_flag;
+    prev_lan_flag = config->lan.flag;
+    config->lan.flag = swaper;
+    config->var_mask = set_mask;
+
+    if (config->var_mask & ETHVAR_MSK(ETHVAR_LAN_FLAGS)) {
+        // if there was a change from STATIC to DHCP
+        if (IS_LAN_STATIC(prev_lan_flag) && IS_LAN_DHCP(config->lan.flag)) {
+            set_LAN_to_dhcp(config);
+        // or STATIC to STATIC
+        } else if (IS_LAN_STATIC(config->lan.flag)){
+            set_LAN_to_static(config);
+        }
+        // from DHCP to DHCP: do nothing
+    }
+
+    save_eth_params(config);
+
+    return 1;
+}
+
+void sntp_get_system_time(char * dest){
+
+  if(sntp_time_init){
+    RTC_TimeTypeDef currTime;
+
+    HAL_RTC_GetTime(&hrtc, &currTime, RTC_FORMAT_BIN);
+
+    snprintf(dest, 10, "%02d:%02d:%02d", currTime.Hours, currTime.Minutes, currTime.Seconds);
```

Comment:
../../src/common/wui_api.c: In function 'sntp_get_system_time':

../../src/common/wui_api.c:237:35: error: '%02d' directive output may be truncated writing between 2 and 3 bytes into a region of size between 2 and 4 [-Werror=format-truncation=]

     snprintf(dest, 10, "%02d:%02d:%02d", currTime.Hours, currTime.Minutes, currTime.Seconds);

                                   ^~~~

../../src/common/wui_api.c:237:24: note: directive argument in the range [0, 255]

     snprintf(dest, 10, "%02d:%02d:%02d", currTime.Hours, currTime.Minutes, currTime.Seconds);

                        ^~~~~~~~~~~~~~~~

../../src/common/wui_api.c:237:5: note: 'snprintf' output between 9 and 12 bytes into a destination of size 10

     snprintf(dest, 10, "%02d:%02d:%02d", currTime.Hours, currTime.Minutes, currTime.Seconds);

     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../src/common/wui_api.c: In function 'sntp_get_system_date':

../../src/common/wui_api.c:250:37: error: 'snprintf' output may be truncated before the last format character [-Werror=format-truncation=]

     snprintf(dest, 12, "%02d.%02d.%d", currDate.Date, currDate.Month + 1, currDate.Year + 1900);

                                     ^

../../src/common/wui_api.c:250:5: note: 'snprintf' output between 11 and 13 bytes into a destination of size 12

     snprintf(dest, 12, "%02d.%02d.%d", currDate.Date, currDate.Month + 1, currDate.Year + 1900);

     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

cc1: all warnings being treated as errors

Reply:
commit  `468c517`